platform/wayland/cog-platform-wl.c

```diff
@@ -618,10 +635,15 @@ pointer_on_button(void              *data,
                   uint32_t           button,
                   uint32_t           state)
 {
-    CogWlPlatform *platform = data;
-    CogWlDisplay  *display = platform->display;
+    CogWlSeat    *seat = data;
+    CogWlDisplay *display = seat->display;
 
-    display->seat.keyboard.serial = serial;
+    if (pointer != seat->pointer_obj) {
+        g_critical("%s: Got pointer %p, expected %p.", G_STRFUNC, pointer, seat->pointer_obj);
+        return;
+    }
+
+    seat->pointer.serial = serial;
```

Comment:
Here we are fixing the wrong use of `display->seat.keyboard.serial = serial;` (https://github.com/Igalia/cog/pull/623/commits/a321b21181571ea271b3465d181e2581e9bc38d7#diff-7342248ee4ad822cd294d9c8341a52fd10b3d1df4dd2a202ccb926df67d51034L624)  for tracking all the serials for the different classes of inputs .  The `cog_wl_seat_get_serial()`gives us the latest serial received (https://github.com/Igalia/cog/pull/623/commits/a321b21181571ea271b3465d181e2581e9bc38d7#diff-b5c8d7e31c380d86e8ae8aea3f2dc76fd5f407590ccc4a6cf2347ca0b9e17749R234).  For example this is important for the grab action in the popup menu: https://github.com/Igalia/cog/pull/623/commits/a321b21181571ea271b3465d181e2581e9bc38d7#diff-7342248ee4ad822cd294d9c8341a52fd10b3d1df4dd2a202ccb926df67d51034R1719xdg 

Ref: https://wayland-book.com/xdg-shell-in-depth/popups.html#popup-grabs